File: src/main/java/com/github/theonlytails/ruby/registries/EnchantmentRegistry.java
package com.github.theonlytails.ruby.registries;

import com.github.theonlytails.ruby.RubyMod;
import com.github.theonlytails.ruby.enchantments.StingerEnchantment;
import net.minecraft.enchantment.Enchantment;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class EnchantmentRegistry {
    public static final DeferredRegister<Enchantment> ENCHANTMENTS = DeferredRegister.create(ForgeRegistries.ENCHANTMENTS, RubyMod.MOD_ID);

    public static final RegistryObject<Enchantment> STINGER = ENCHANTMENTS.register("stinger", StingerEnchantment::new);
}

File: src/main/kotlin/com/github/theonlytails/rubymod/registries/EnchantmentRegistry.kt
@file:Suppress("unused")

package com.github.theonlytails.rubymod.registries

import com.github.theonlytails.rubymod.RubyMod
import com.github.theonlytails.rubymod.enchantments.StingerEnchantment
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

object EnchantmentRegistry {
	val ENCHANTMENTS =
		KDeferredRegister(ForgeRegistries.ENCHANTMENTS, RubyMod.MOD_ID)

	val STINGER by ENCHANTMENTS.register("stinger", ::StingerEnchantment)
}